### PR TITLE
userdb: various userdb fixes and minor feature additions

### PR DESCRIPTION
**Pull request by poettering**
Fixes: #21215 #21222 #21223

### CODE REVIEW

File: src/userdb/userwork.c [hunk 1]
```diff
@@ -153,7 +152,7 @@ static int vl_method_get_user_record(Varlink *link, JsonVariant *parameters, Var
                 return r;
 
         r = userdb_flags_from_service(link, p.service, &userdb_flags);
-        if (r < 0)
+        if (r != 0)
```
Comment (bluca) [inline src/userdb/userwork.c]: We usually do `< 0` and the function returns either 0 or `varlink_error()` so this is a bit confusing to the reader, maybe a comment to explain it?

**Review by bluca — APPROVED** [review]
lgtm, just one nit

**Comment by poettering** [comment]
added the suggested comment, upgrading green label

**Pull request merged**